src/window/building/common.c

```diff
@@ -53,24 +53,34 @@ int window_building_get_vertical_offset(building_info_context *c, int new_window
     return new_window_y;
 }
 
-void window_building_draw_employment(building_info_context *c, int y_offset)
+void window_building_draw_employment(building_info_context* c, int y_offset)
 {
-    building *b = building_get(c->building_id);
+    do_window_building_draw_employment(c, y_offset, 1);
+}
+
+void do_window_building_draw_employment(building_info_context* c, int y_offset, int consider_house_covering)
+{
+    building* b = building_get(c->building_id);
     int text_id;
     if (b->num_workers >= model_get_building(b->type)->laborers) {
         text_id = 0;
-    } else if (city_population() <= 0) {
+    }
+    else if (city_population() <= 0) {
         text_id = 16; // no people in city
-    } else if (b->houses_covered <= 0) {
+    }
+    else if (consider_house_covering == 1 && b->houses_covered <= 0) {
```

Comment:
The `consider_house_covering` variable is a boolean: just use `if (consider_house_covering)`, instead of checking for `== 1`.
Also, instead of checking for this variable every single `if` statement, you can also add the following code before this if-statement:
```
} else if (!consider_house_covering) {
   text_id = 19; // too few people allocated
} else ... the rest of the if statements
```

Reply:
Ok